---
title: "Simulate traits"
author: "Frederick J. Boehm"
date: "9/25/2019"
output: html_document
---

## Read DO data

```{r saveRDS}
if (!file.exists(here::here("data", "derived_data", "genoprobs.rds"))){
  download.file("https://datadryad.org/stash/downloads/file_stream/63306",
              destfile = here::here("data", "raw_data", "Attie_DO378_eQTL_viewer_v1.Rdata")
              )
  load(here::here("data", "raw_data", "Attie_DO378_eQTL_viewer_v1.Rdata"))
  saveRDS(K, here::here("data", "derived_data", "kinship.rds"))
  saveRDS(map, here::here("data", "derived_data", "map.rds"))
  saveRDS(genoprobs, here::here("data", "derived_data", "genoprobs.rds"))
  saveRDS(dataset.islet.rnaseq$covar, here::here("data", "derived_data", "covar.rds"))
}
```

```{r readRDS}
probs <- readRDS(here::here("data", "derived_data", "genoprobs.rds"))
K <- readRDS(here::here("data", "derived_data", "kinship.rds"))
covar <- readRDS(here::here("data", "derived_data", "covar.rds"))
map <- readRDS(here::here("data", "derived_data", "map.rds"))
```

## Read CC genotype probabilities

```{r load-pkgs}
library(tidyverse)
```

```{r read-cc}
files <- dir(here::here("data", "raw_data", "CC-genoprobs"))[1:3]
genoprob_cc <- purrr::map(.x = here::here("data", "raw_data", "CC-genoprobs", files), .f = read_csv) %>%
  purrr::map(.f = function(x){array(data = x, dim = c(1, 39, 76689))})
genoprob_cc %>%
dplyr::bind_rows()
```
```{r}
load(here::here("data", "derived_data", "sassetti",  "rqtl2_28Sep2018_mapping_objects.Rdata"))
```

## Simulate traits from CC genotypes

We'll simulate 1000 traits (for each CC line and each DO mouse) at each combination of levels. 

What are the experimental design factors?

1. Signal strength
1. allele effects partitioning

We'll then map the resulting traits in both the 52 CCs and the 378 DO mice separately. Can we detect QTL in both CCs and in DOs?



